components/grid/state.md

```diff
@@ -10,923 +10,419 @@ position: 50
-
-To reset the grid state, call `SetStateAsync(null)`.
-
-You should avoid calling `SetStateAsync` in the grid [CRUD methods]({%slug components/grid/editing/overview%}) (such as [OnRead]({%slug components/grid/manual-operations%}), `OnUpdate`, `OnEdit`, `OnCreate`, `OnCancel`). Doing so may lead to unexpected results because the grid has more logic to execute after the event.
-
 ## Information in the Grid State
 
-The following information is present in the grid state:
+The Grid state is a generic [class `GridState<TItem>`](/blazor-ui/api/Telerik.Blazor.Components.GridState-1). The type depends on the type of the Grid model. The `GridState<TItem>` object exposes the following properties:
 
-* **Editing** - whether the user was inserting or editing an item (opens the same item for editing with the current data from the built-in editors of the grid - the data is updated in the `OnChange` event, not on every keystroke for performance reasons). The `OriginalEditItem` carries the original model without the user modifications so you can compare.
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
 
-* **Filtering** - filter descriptors (fields by which the grid is filtered, the operator and value).
+| Property | Type | Description |
+| --- | --- | --- |
+| `CollapsedGroups` | `ICollection<int>` | The indexes of all collapsed [groups]({%slug components/grid/features/grouping%}) when `LoadGroupsOnDemand="false"`, i.e. when not [loading groups on demand]({%slug grid-group-lod%}). |
+| `ColumnStates` | `ICollection<GridColumnState>` | Information about each [column's reorder index, width, visibility, locked state, `Id` parameter value and `Field`]({%slug components/grid/columns/bound%}). The column order in the collection matches the column order in the Grid declaration. On the other hand, the `Index` property matches the current column's position in the UI. **`Id` and `Field` are always `null` after deserialization, because these properties have no public setters.** |
+| `EditField` | `string` | The currently edited data item property in [`Incell` edit mode]({%slug components/grid/editing/incell%}). |
+| `EditItem` | `TItem`* | The currently edited data item in [any edit mode]({%slug components/grid/editing/overview%}). |
+| `ExpandedItems` | `ICollection<TItem>` | The expanded data items, when [using `<DetailTemplate>` (hierarchy)]({%slug components/grid/features/hierarchy%}). |
+| `FilterDescriptors` | `ICollection<IFilterDescriptor>` | All filtering criteria, except the ones that relate to the[`GridSearchBox`]({%slug grid-searchbox%}). |
+| `GroupDescriptors` | `ICollection<GroupDescriptor>` | Information about currently applied [grouping]({%slug components/grid/features/grouping%}). |
+| `InsertedItem` | `TItem`* | The data item that is being added in `Inline` or `Popup` edit mode. |
```

Comment:
For some people, it may be unclear why this property is not available in Incell edit mode and questions may arise. I suggest adding a note here and/or link to the[ Incell specifics](https://docs.telerik.com/blazor-ui/components/grid/editing/incell#event-sequence)